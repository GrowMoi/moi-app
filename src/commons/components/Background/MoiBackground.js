import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { ImageBackground } from 'react-native';
import { connect } from 'react-redux';
import styled from 'styled-components/native';
import { LANDSCAPE } from '../../../constants';

const Background = styled(ImageBackground)`
  width: ${props => props.width};
  height: ${props => props.height};
  flex: 1;
`;

@connect(store => ({ device: store.device }))
export default class MoiBackground extends Component {
  render() {
    const { device, style } = this.props;
    const { width, height, orientation } = device.dimensions;
    const currentImage = orientation === LANDSCAPE ? 'background_tree_landscape' : 'background_tree_portrait';
    return (
      <Background style={style} width={width} height={height} source={{uri: currentImage}} resizeMode='cover'>
        {this.props.children}
      </Background>
    );
  }
}

MoiBackground.propTypes = {
  children: PropTypes.any,
  device: PropTypes.object,
  style: PropTypes.any,
};
