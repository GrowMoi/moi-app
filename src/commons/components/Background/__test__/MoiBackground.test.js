import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components/native';

import MoiBackground from '../MoiBackground';

const initialState = {
  dimensions: {
    width: 320,
    height: 568,
    orientation: 'PORTRAIT',
  },
};

describe('MoiBackground', () => {
  it('should render expected', () => {
    const wrapper = renderer.create(<MoiBackground device={initialState} />);
    wrapper.toJSON();

    expect(wrapper).toMatchSnapshot();
    expect(wrapper).toBeTruthy();
  });
});
