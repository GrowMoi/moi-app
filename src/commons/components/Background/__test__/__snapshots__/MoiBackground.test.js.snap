// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoiBackground should render expected 1`] = `
<Image
  height={568}
  resizeMode="cover"
  source={1}
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "height": 568,
        "width": 320,
      },
      undefined,
    ]
  }
  width={320}
/>
`;
